File: frontend/src/microservices/content/objects/BlogPostResponse.ts
// 本文件定义 content 模块的 `BlogPostResponse`，作为响应数据并提供 JSON 编解码。

import type { BlogCommentResponse } from './BlogCommentResponse'
import type { BlogPostSummaryResponse } from './BlogPostSummaryResponse'

export type BlogPostResponse = {
  post: BlogPostSummaryResponse
  content: string
  comments: BlogCommentResponse[]
}
export const blogPostResponseFromJson = (json: string): BlogPostResponse =>
  JSON.parse(json) as BlogPostResponse

export const blogPostResponseToJson = (value: BlogPostResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/blog/objects/BlogPostResponse.scala
// BlogPostResponse：博客域博客文章返回对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BlogPostResponse(post: BlogPostSummaryResponse, content: String, comments: List[BlogCommentResponse])
object BlogPostResponse:
  given sourceEncoder: Encoder[BlogPostResponse] = deriveEncoder
  given sourceDecoder: Decoder[BlogPostResponse] = deriveDecoder